spec/mongo/auth/stringprep_spec.rb

```diff
@@ -0,0 +1,202 @@
+require 'spec_helper'
+
+describe Mongo::Auth::StringPrep do
+  include Mongo::Auth::StringPrep
+
+  describe '#prepare' do
+    let(:prepared_data) do
+      prepare(data, mappings, prohibited, options)
+    end
+
+    context 'with no options' do
+      let(:mappings) do
+        []
+      end
+
+      let(:prohibited) do
+        []
+      end
+
+      let(:options) do
+        {}
+      end
+
+      context 'when the data has invalid bidi' do
+        let(:data) do
+          "\u0627\u0031"
+        end
+
+        it 'does not raise an error' do
+          expect(prepared_data).to eq("\u0627\u0031")
+        end
+      end
+
+      context 'when the data has unicode codes' do
+        let(:data) do
+          "ua\u030Aer"
+        end
+
+        it 'does not normalize the data' do
+          expect(prepared_data).to eq("ua\u030Aer")
+        end
+      end
+    end
+
+    context 'with options specified' do
+      let (:mappings) do
+        [Mongo::Auth::StringPrep::Tables::B1, Mongo::Auth::StringPrep::Tables::B2]
+      end
+
+      let (:prohibited) do
+        [
+          Mongo::Auth::StringPrep::Tables::C1_1,
+          Mongo::Auth::StringPrep::Tables::C1_2,
+          Mongo::Auth::StringPrep::Tables::C6,
+        ]
+      end
```

Comment:
I'm curious about how we chose to test these mappings and prohibited list and not other subsets?

Reply:
I copied this from David's Go implementation of stringprep. I think the idea is more to test that the `prepare` function properly handles the `prohibited` argument rather than to test that the tables contain the data we expect.